test/Tests/Services/ISqlExecutorTests.cs

```diff
@@ -0,0 +1,172 @@
+
+            Assert.Multiple(() =>
+            {
+                connection.Received().Open();
+                command.Received().ExecuteReader();
+                Assert.That(command.CommandText, Is.EqualTo(SCRIPT_TEXT));
+                Assert.That(data.Count(), Is.EqualTo(2));
+                Assert.That(data.FirstOrDefault()?.Property(COL_1)?.Value.ToString(), Is.EqualTo(VAL_1));
+                Assert.That(data.FirstOrDefault()?.Property(COL_2)?.Value.ToString(), Is.EqualTo(VAL_2));
+
+            });
+        }
+
+
+        [Test]
+        public async Task ExecuteQuery_InvalidScript_Throws()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+            string queryName = "InvalidScript.sql";
+
+            try
+            {
+                await sqlExecutor.ExecuteQuery(CONN_STRING, queryName);
+
+                Assert.Multiple(() =>
+                {
+                    connection.DidNotReceive().Open();
+                    command.DidNotReceive().ExecuteReader();
+                    Assert.Fail($"Resource '{queryName}' not found.");
+                });
+            }
+            catch (Exception) { }
+        }
+
+
+        [Test]
+        public async Task ExecuteNonQuery_ValidScript_Executes()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+
+            await sqlExecutor.ExecuteNonQuery(CONN_STRING, VALID_SCRIPT);
+
+            Assert.Multiple(() =>
+            {
+                connection.Received().Open();
+                command.Received().ExecuteNonQuery();
+                Assert.That(command.CommandText, Is.EqualTo(SCRIPT_TEXT));
+
+            });
+        }
+
+
+        [Test]
+        public async Task ExecuteNonQuery_WithParameters_AddsParameters()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+            string param1Name = "@K1";
+            string param1Value = "V1";
+            Dictionary<string, object> parameters = new()
+            {
+                { param1Name, param1Value }
+            };
+
+            await sqlExecutor.ExecuteNonQuery(CONN_STRING, VALID_SCRIPT, parameters);
+
+            Assert.Multiple(() =>
+            {
+                connection.Received().Open();
+                command.Received().ExecuteNonQuery();
+                Assert.That(parameter.ParameterName, Is.EqualTo(param1Name));
+                Assert.That(parameter.Value, Is.EqualTo(param1Value));
+
+            });
+        }
+
+
+        [Test]
+        public async Task ExecuteNonQuery_InvalidScript_Throws()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+            string queryName = "InvalidScript.sql";
+
+            try
+            {
+                await sqlExecutor.ExecuteNonQuery(CONN_STRING, queryName);
```

Comment:
This is dangerous as the Assert.Multiple throws an exception in case of failure.
It means that assert fail won't fail the test as the exception is caught.
Use Assert.Throws or Assert.ThrowsAsync instead.